doc/en/user/source/production/fips.rst

```diff
@@ -0,0 +1,278 @@
+
+The BCFKS format provides FIPS 140-2 compliance:
+
+.. code-block:: properties
+
+   GEOSERVER_KEYSTORE_TYPE=BCFKS
+   GEOSERVER_KEYSTORE_PROVIDER=BC
+
+JCEKS (Java Cryptography Extension KeyStore)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Traditional Java keystore format for backward compatibility:
+
+.. code-block:: properties
+
+   GEOSERVER_KEYSTORE_TYPE=JCEKS
+   GEOSERVER_KEYSTORE_PROVIDER=SunJCE
+
+PKCS12
+~~~~~~
+
+Standard format for certificate and key storage:
+
+.. code-block:: properties
+
+   GEOSERVER_KEYSTORE_TYPE=PKCS12
+   GEOSERVER_KEYSTORE_PROVIDER=BC
+
+Migration from JCEKS to BCFKS
+-----------------------------
+
+GeoServer provides a migration utility to convert existing JCEKS keystores to FIPS-compliant BCFKS format.
+
+Using the Migration Utility
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The migration utility is provided as a standalone script that uses the FIPS utilities JAR:
+
+.. code-block:: bash
+
+   ./migrate-keystore.sh -s source.jks -sp oldpass -t target.bcfks -tp newpass
+
+The script automatically handles:
+* Dependency checking (BouncyCastle JARs)
+* File validation and permissions
+* Auto-detection of keystore types and providers
+* Error handling and colored output
+
+Parameters:
+
+* ``-s, --source``: Source keystore file path
+* ``-sp, --source-password``: Source keystore password
+* ``-t, --target``: Target keystore file path
+* ``-tp, --target-password``: Target keystore password
+* ``-st, --source-type``: Source keystore type (auto-detected if not specified)
+* ``-spr, --source-provider``: Source keystore provider (auto-detected if not specified)
+* ``-tt, --target-type``: Target keystore type (default: BCFKS)
+* ``-tpr, --target-provider``: Target keystore provider (auto-detected if not specified)
+* ``-v, --verbose``: Verbose output
+* ``-h, --help``: Show help message
+
+Automated Migration
+~~~~~~~~~~~~~~~~~~
+
+Use the provided test script for automated migration testing:
+
+.. code-block:: bash
+
+   ./test-fips-docker.sh
+
+This script will:
+
+1. Build the FIPS-enabled GeoServer container
+2. Create test keystores (JCEKS and BCFKS)
+3. Test migration between formats
+4. Validate FIPS mode activation
+5. Test security endpoints
+
+Standalone Utilities
+~~~~~~~~~~~~~~~~~~~
+
+The FIPS utilities are provided as a separate module that is not deployed into the main GeoServer application:
+
+* **Security**: Migration utilities are not part of the production runtime
+* **Flexibility**: Can be used independently of GeoServer
+* **Maintenance**: Utilities can be updated without affecting the main application
+
+The utilities include:
+
+* **Migration Script**: ``migrate-keystore.sh`` - User-friendly keystore migration
+* **FIPS Utilities JAR**: ``fips-utils.jar`` - Core migration and validation logic
+* **Test Script**: ``test-fips-docker.sh`` - Automated testing and validation
+
+Verifying FIPS Mode
+------------------
+
+Check that FIPS mode is active by examining the GeoServer logs:
```

Comment:
somewhere in here I think it should mention that FIPS mode is an OS level setting, and to fully test that GeoServer is running in a FIPS mode environment, you need to enable it.